gcp_variant_transforms/beam_io/vcf_parser.py

```diff
@@ -301,18 +296,17 @@ def __init__(self,
               self._process_header_lines),
           **kwargs)
 
-    if isinstance(self, PySamParser):
-      # In an absence of "#CHROM POS..." header line, PySam process would lock.
-      # However, we need to support cases when no records in files were provided
-      # even if the above header line is missing. To circumvent the error, read
-      # the file until the first line is found, and verify validity.
-      self._header_lines = None
-      if isinstance(range_tracker, range_trackers.UnsplittableRangeTracker):
-        secondary_tracker = range_trackers.UnsplittableRangeTracker(
-            copy(range_tracker._range_tracker))
-      else:
-        secondary_tracker = copy(range_tracker)
-      self._verify_header(copy(text_source), secondary_tracker)
+    # In an absence of "#CHROM POS..." header line, PySam process would lock.
+    # However, we need to support cases when no records in files were provided
+    # even if the above header line is missing. To circumvent the error, read
+    # the file until the first line is found, and verify validity.
+    self._header_lines = None
+    if isinstance(range_tracker, range_trackers.UnsplittableRangeTracker):
+      secondary_tracker = range_trackers.UnsplittableRangeTracker(
+          copy(range_tracker._range_tracker))
+    else:
+      secondary_tracker = copy(range_tracker)
+    self._verify_header(copy(text_source), secondary_tracker)
 
     self._text_lines = text_source.read_records(self._file_name,
                                                 range_tracker)
```

Comment:
This whole class needs to be reworked to only have a single class (since we are removing other parses completely), but I was going to do it later. Would you like me to do it in this PR?